src/settings.h

```diff
@@ -111,6 +111,15 @@ class Settings
         return ms_registryPath;
     }
 
+    /// Return DSA public key to verify update file signature
+    static std::string GetDSAPubKeyPem()
+    {
+        CriticalSectionLocker lock(ms_csVars);
+        if (ms_DSAPubKey.empty())
+            SetDSAPubKeyPem(GetCustomResource("DSAPub", "DSAPEM"));
```

Comment:
I know (now…took me writing first version of this comment to realize it) that this works, because Windows critical sections are re-entrant, but I think it's not very intuitive — I expect this to deadlock because `SetDSAPubKeyPem` also holds `ms_csVars`.

I would prefer to do this in the same way as `GetAppcastURL()` above, i.e. just assign into `ms_DSAPubKey`. This  would have consequences for other code, of course — I'll comment on them in their respective places.